bindings/python/crocoddyl/multibody/costs/cost-sum.cpp

```diff
@@ -68,14 +73,20 @@ void exposeCostSum() {
                                                         "Initialize the total cost model.\n\n"
                                                         "For this case the default nu is equals to model.nv.\n"
                                                         ":param state: state of the multibody system"))
-      .def("addCost", &CostModelSum::addCost, bp::args("self", "name", "cost", "weight"),
-           "Add a cost item.\n\n"
-           ":param name: cost name\n"
-           ":param cost: cost model\n"
-           ":param weight: cost weight")
+      .def("addCost", &CostModelSum::addCost,
+           CostModelSum_addContact_wrap(bp::args("self", "name", "cost", "weight", "active"),
+                                        "Add a cost item.\n\n"
+                                        ":param name: cost name\n"
+                                        ":param cost: cost model\n"
+                                        ":param weight: cost weight\n"
+                                        ":param active: True if the cost is activated (default true)"))
       .def("removeCost", &CostModelSum::removeCost, bp::args("self", "name"),
            "Remove a cost item.\n\n"
            ":param name: cost name")
+      .def("changeCostStatus", &CostModelSum::changeCostStatus, bp::args("self", "name", "active"),
+           "Change the cost status.\n\n"
```

Comment:
Can't we deduce `active` or `inactive` from whether the `weight==0`?

Reply:
No, we can't.

We could still run computation (`active=true`) with a `weight=0.`.

It is possible to adopt the policy of disabling the computation (`active=false`) whenever we add a cost with `weight=0.`, however this might produce user errors if this later set a weight different to zero, e.g.
```cpp
sum_cost.addCost("my_cost", cost, 0., true); // if we adopt this policy, then `active=false` without any user acknowledgment
sum_cost.get_costs()->second("my_cost")->weight = 1.;
```

So I am not sure we should adopt this policy, what do you think?

Reply:
Okay, I see. So the question becomes what happens if `nr_` changes?

Reply:
If the cost is disable, then the total `nr_` is decreased. The same for the activating a cost.

There is an unittest code that checks this. See this: https://github.com/cmastalli/crocoddyl/blob/topic/cost-status/unittest/test_cost_sum.cpp#L52

Note that `nr_` (for cost sum) represents the total `nr_`.

Reply:
@wxmerkt do you have any further question or request?

Reply:
Yes, I saw that it is decreased. Does that change allocation or matrix sizes? Is another operation required after activating/deactivating to avoid data corruption or sth therelike?

Reply:
This is a good point, let me add some relevant information about it.

`nr` is not use for data allocation inside cost sum, however, it is used for numdiff cost sum.

Therefore, we would expect an error if we change the cost status and use numdiff to retrieve calcDiff.

To solve this, we should have to kind of `nr`: one for the current `nr` another one for the total `nr`. This latter one is only used to allocate data for numdiff. Concretely, I proposed a new function call `get_nr_total()`.

**Further notes**
We might have a similar problem for the contact and impulse status. I will handle this issue in this PR too, and will also include other unittest code for checking it.

Reply:
Thank you for the detailed explanation and highlighting that this issue is (luckily) limited to the num-diff cases. I think we should use a more descriptive name for `nra`, otherwise your proposal sounds great and clarifies all my questions :-)

Reply:
What about the above options?
 1. `get_nr_allocated`
 2. `get_nr_total`

any preference?

Reply:
(2) sounds good :-) `allocated` is a concept that may need further explanation